pkg/build/builder/cmd/builder_test.go

```diff
@@ -0,0 +1,266 @@
+				"https.proxy": fakeHTTPSProxy,
+			},
+		},
+		{
+			name: "git clone proxy",
+			build: &buildv1.Build{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-build-1",
+					Namespace: "test",
+				},
+				Spec: buildv1.BuildSpec{
+					CommonSpec: buildv1.CommonSpec{
+						Source: buildv1.BuildSource{
+							Type: buildv1.BuildSourceGit,
+							Git: &buildv1.GitBuildSource{
+								URI: "https://githost.dev/myorg/myrepo.git",
+							},
+							SourceSecret: &corev1.LocalObjectReference{
+								Name: "git-auth",
+							},
+						},
+					},
+				},
+			},
+		},
+		{
+			name: "git clone proxy auth",
+			build: &buildv1.Build{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-build-1",
+					Namespace: "test",
+				},
+				Spec: buildv1.BuildSpec{
+					CommonSpec: buildv1.CommonSpec{
+						Source: buildv1.BuildSource{
+							Type: buildv1.BuildSourceGit,
+							Git: &buildv1.GitBuildSource{
+								URI: "https://githost.dev/myorg/myrepo.git",
+								ProxyConfig: buildv1.ProxyConfig{
+									HTTPProxy:  &fakeHTTPProxy,
+									HTTPSProxy: &fakeHTTPSProxy,
+								},
+							},
+							SourceSecret: &corev1.LocalObjectReference{
+								Name: "git-auth",
+							},
+						},
+					},
+				},
+			},
+			expectedLocalConfigs: map[string]string{
+				"http.proxy":  fakeHTTPProxy,
+				"https.proxy": fakeHTTPSProxy,
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			builder := &builderConfig{}
+			builder.build = tc.build
+			gitClient := newFakeGitRepository()
+			err := builder.setupProxyConfig(gitClient, "test")
+			if err != nil && !tc.expectError {
+				t.Errorf("received unexpected error: %v", err)
+			}
+			if err == nil && tc.expectError {
+				t.Error("expected to receive error, got none")
+			}
+
+			if len(tc.expectedLocalConfigs) != len(gitClient.LocalConfigs) {
+				t.Errorf("expected local configs to have %d entries, got %d", len(tc.expectedLocalConfigs), len(gitClient.LocalConfigs))
+			}
+
+			if len(tc.expectedGlobalConfigs) != len(gitClient.GlobalConfigs) {
+				t.Errorf("expected global configs to have %d entries, got %d", len(tc.expectedGlobalConfigs), len(gitClient.GlobalConfigs))
+			}
+
+			for config, expected := range tc.expectedLocalConfigs {
+				actual, present := gitClient.LocalConfigs[config]
+				if !present {
+					t.Errorf("expected local git configuration %s was not set", config)
+				}
+				if actual != expected {
+					t.Errorf("expected local git config %s to be %s, got %s", config, expected, actual)
+				}
+			}
+			for config, expected := range tc.expectedGlobalConfigs {
+				actual, present := gitClient.GlobalConfigs[config]
+				if !present {
+					t.Errorf("expected global git configuration %s was not set", config)
+				}
+				if actual != expected {
+					t.Errorf("expected global git config %s to be %s, got %s", config, expected, actual)
+				}
+			}
+		})
+	}
+
+}
+
+func newFakeGitRepository() *fakeGitRepository {
```

Comment:
nit: This and below should possibly be factored out into a test framework like we had for the ciro https://github.com/openshift/cluster-image-registry-operator/tree/master/test/framework

Reply:
updated - moved to a "fake" package since it's used for unit testing.